sensors/Dht.py

```diff
@@ -0,0 +1,98 @@
+# inheritance library
+# import abc
+import Adafruit_DHT
+# GPIO lib
+import RPi.GPIO as GPIO           # import RPi.GPIO module
+
+from sensors.SensorException import SensorReadError
+
+
+# SensorInterface(Top Hierarchy)
+# Author     : clintonbf
+# Repository :   https://github.com/clintonbf/sensor_interface/blob/master/sensors/SensorInterface.py
+from sensors.Sensor import Sensor
+
+# unit conversion function
+# Author     : clintonbf
+# Repository : https://github.com/clintonbf/sensor_interface/tree/master/unit_conversion
+from utils.unit_conversion.temperature import celsius_to_kelvin
+
+DATA_INDICES = {
+    "relative_humidity": 0,
+        "temperature": 1,
+    }
+
+class Dht(Sensor):
+    """
+        Credit for connect_to_port and get_raw_data goes to
+        Noah MacRitchie (noah21mac@gmail.com) and Andrey Goryelov (andrey.goryelov@gmail.com)
+    """
+
+    # TODO: what is timeout ?
+    # TODO: It is just a GPIO data in, not serial connection.
+    # TODO: pass it in like : gpio_in = 1, named params
+   
+    def __init__(self, uid: int, gpio_in: str, model : str ) -> None:
+        # super().__init__()
+        self.__uid = uid
+        self.__gpio_in = gpio_in
+        # self.__timeout = timeout
+        if model.lower() == 'dht11':
+            self.__model = 11
+        # elif model.lower() == 'am2302':
+        #     self.__model = 22
+        else:
+            self.__model = 22
+        self.__reading = () # TODO meaning?
+
+    @classmethod
```

Comment:
Remove this method; it's been redefined in SensorDataFormatter

Reply:
When being instantiated, the constructor receives a string of the model, but the Adafruit library only recognizes numbers, so you need to pass in 22 for DHT22. I don't see how this is redefined in SensorDataFormatter

Reply:
Sorry, meant the block below (lines 49 - 52)

Reply:
Removed, good point.